initfiles/componentfiles/configschema/xsd/esp.xsd

```diff
@@ -145,91 +214,176 @@
-                                <xs:element name="CSR" type="xs:string" minOccurs="0" hpcc:displayName="Certificate Signing Request" hpcc:tooltip="The generated Certificate Signing Request (CSR) is displayed here after first deployment" />
+                                <xs:element name="Certificate" type="xs:string" minOccurs="0"
+                                            hpcc:displayName="Certificate"
+                                            hpcc:tooltip="A certificate can be pasted here from the clipboard.  Otherwise, the generated certificate is displayed here after deployment"/>
+                                <xs:element name="PrivateKey" type="xs:string" minOccurs="0"
+                                            hpcc:displayName="Private Key"
+                                            hpcc:tooltip="A private key can be pasted here from the clipboard.  Otherwise, the generated private key is displayed here after deployment"/>
+                                <xs:element name="CSR" type="xs:string" minOccurs="0"
+                                            hpcc:displayName="Certificate Signing Request"
+                                            hpcc:tooltip="The generated Certificate Signing Request (CSR) is displayed here after first deployment"/>
                             </xs:sequence>
                             <xs:attributeGroup ref="computerNodeReference"/>
-                            <xs:attribute name="directory" type="absolutePath" use="optional" hpcc:hidden="true" />
-                            <xs:attribute name="FQDN" type="xs:string" use="optional" hpcc:displayName="Fully Qualified Domain Name" hpcc:tooltip="Enter the fully qualified domain name" />
+                            <xs:attribute name="directory" type="absolutePath" use="optional" hpcc:hidden="true"/>
+                            <xs:attribute name="FQDN" type="xs:string" use="optional"
+                                          hpcc:displayName="Fully Qualified Domain Name"
+                                          hpcc:tooltip="Enter the fully qualified domain name"/>
                         </xs:complexType>
                     </xs:element>
-                    <xs:element name="Notes" type="usernotes"  hpcc:displayName="Notes"/>
+                    <xs:element name="Notes" type="usernotes" hpcc:displayName="Notes"/>
                 </xs:sequence>
 
                 <xs:attributeGroup ref="buildInfo"/>
-                <xs:attribute name="name" type="xs:string" use="required" hpcc:displayName="Name" hpcc:autoGenerateType="prefix_" hpcc:autoGenerateValue="myesp"
+                <xs:attribute name="name" type="xs:string" use="required" hpcc:displayName="Name"
+                              hpcc:autoGenerateType="prefix_" hpcc:autoGenerateValue="myesp"
```

Comment:
We need to document hpcc:autoGenerateType, hpcc:autoGenerateValue, and other hpcc:... .

Reply:
These are documented already in the Wiki: http://10.240.34.10/wiki/index.php/XSD_Schema_Definition